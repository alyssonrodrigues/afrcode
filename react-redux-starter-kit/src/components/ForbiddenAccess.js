import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { withStyles } from '@material-ui/core/styles'
import Dialog from '@material-ui/core/Dialog'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContenText from '@material-ui/core/DialogContentText'
import DialogActions from '@material-ui/core/DialogActions'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'

import { dismissMessage } from '../actions/messagesActions'

const styles = {
  paperClassName: {
    borderRadius: '2%',
    outline: 'none'
  },
  backdropClassName: {
    backgroundColor: 'rgba(255, 255, 255, 0.65)'
  },
  item: {
    paddingTop: 12
  },
  link: {
    padding: 12
  }
}

export class ForbiddenAccess extends Component {
  componentWillUnmount () {
    this.props.dismissMessage()
  }

  render () {
    const { classes, history } = this.props
    return (
      <main>
        <Dialog open
          BackdropProps={{className: classes.backdropClassName}}
          PaperProps={{className: classes.paperClassName, elevation: 2}}>
          <DialogTitle disableTypography>
            <Typography variant='display1' color='error'>
              Acesso negado
            </Typography>
          </DialogTitle>
          <DialogContent>
            <DialogContenText>
              <Typography variant='title' style={{ maxWidth: 500 }}>
                Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.
              </Typography>
            </DialogContenText>
            <DialogContenText>
              <Typography variant='subheading' className={classes.item}>
                Entre em contato através do 0800 644 1500 se o problema persistir.
              </Typography>
            </DialogContenText>
          </DialogContent>
          <DialogActions>
            <Button className={classes.link} onClick={() => history.push('/')} color='primary'>
              Voltar
            </Button>
          </DialogActions>
        </Dialog>
      </main>
    )
  }
}

const materialUIEnhanced = withStyles(styles)(ForbiddenAccess)

const mapStateToProps = ({ messages }) => ({ messages })

const reduxEnhanced = connect(mapStateToProps, { dismissMessage })(materialUIEnhanced)

export default withRouter(reduxEnhanced)
