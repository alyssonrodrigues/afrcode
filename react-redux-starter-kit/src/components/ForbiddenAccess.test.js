import React from 'react'
import { shallow } from 'enzyme'

import { ForbiddenAccess } from './ForbiddenAccess'

describe('ForbiddenAccess', () => {
  it('ForbiddenAccess render deveria ocorrer sem falhas!', () => {
    const componentWrapper = shallow(
      <ForbiddenAccess classes={{}} dismissMessage={jest.fn()} />
    )
    expect(componentWrapper).toMatchSnapshot()
  })
})
