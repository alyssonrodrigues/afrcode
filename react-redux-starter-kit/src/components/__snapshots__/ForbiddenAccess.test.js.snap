// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForbiddenAccess ForbiddenAccess render deveria ocorrer sem falhas! 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForbiddenAccess
    classes={Object {}}
    dismissMessage={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Dialog)
        BackdropProps={
                Object {
                        "className": undefined,
                      }
        }
        PaperProps={
                Object {
                        "className": undefined,
                        "elevation": 2,
                      }
        }
        open={true}
>
        <WithStyles(DialogTitle)
                disableTypography={true}
        >
                <WithStyles(Typography)
                        color="error"
                        variant="h4"
                >
                        Acesso negado
                </WithStyles(Typography)>
        </WithStyles(DialogTitle)>
        <WithStyles(DialogContent)>
                <WithStyles(DialogContentText)
                        component="div"
                >
                        <WithStyles(Typography)
                                style={
                                        Object {
                                                "maxWidth": 500,
                                              }
                                }
                                variant="h6"
                        >
                                Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.
                        </WithStyles(Typography)>
                </WithStyles(DialogContentText)>
                <WithStyles(DialogContentText)
                        component="div"
                >
                        <WithStyles(Typography)
                                className={undefined}
                                variant="subtitle1"
                        >
                                Entre em contato através do 0800 644 1500 se o problema persistir.
                        </WithStyles(Typography)>
                </WithStyles(DialogContentText)>
        </WithStyles(DialogContent)>
        <WithStyles(DialogActions)>
                <WithStyles(Button)
                        className={undefined}
                        color="primary"
                        onClick={[Function]}
                >
                        Voltar
                </WithStyles(Button)>
        </WithStyles(DialogActions)>
</WithStyles(Dialog)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "BackdropProps": Object {
          "className": undefined,
        },
        "PaperProps": Object {
          "className": undefined,
          "elevation": 2,
        },
        "children": Array [
          <WithStyles(DialogTitle)
            disableTypography={true}
>
            <WithStyles(Typography)
                        color="error"
                        variant="h4"
            >
                        Acesso negado
            </WithStyles(Typography)>
</WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)
                        component="div"
            >
                        <WithStyles(Typography)
                                    style={
                                                Object {
                                                            "maxWidth": 500,
                                                          }
                                    }
                                    variant="h6"
                        >
                                    Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.
                        </WithStyles(Typography)>
            </WithStyles(DialogContentText)>
            <WithStyles(DialogContentText)
                        component="div"
            >
                        <WithStyles(Typography)
                                    className={undefined}
                                    variant="subtitle1"
                        >
                                    Entre em contato através do 0800 644 1500 se o problema persistir.
                        </WithStyles(Typography)>
            </WithStyles(DialogContentText)>
</WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
                        className={undefined}
                        color="primary"
                        onClick={[Function]}
            >
                        Voltar
            </WithStyles(Button)>
</WithStyles(DialogActions)>,
        ],
        "open": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography)
              color="error"
              variant="h4"
>
              Acesso negado
</WithStyles(Typography)>,
            "disableTypography": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Acesso negado",
              "color": "error",
              "variant": "h4",
            },
            "ref": null,
            "rendered": "Acesso negado",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(DialogContentText)
                component="div"
>
                <WithStyles(Typography)
                                style={
                                                Object {
                                                                "maxWidth": 500,
                                                              }
                                }
                                variant="h6"
                >
                                Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.
                </WithStyles(Typography)>
</WithStyles(DialogContentText)>,
              <WithStyles(DialogContentText)
                component="div"
>
                <WithStyles(Typography)
                                className={undefined}
                                variant="subtitle1"
                >
                                Entre em contato através do 0800 644 1500 se o problema persistir.
                </WithStyles(Typography)>
</WithStyles(DialogContentText)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)
                  style={
                                    Object {
                                                      "maxWidth": 500,
                                                    }
                  }
                  variant="h6"
>
                  Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.
</WithStyles(Typography)>,
                "component": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.",
                  "style": Object {
                    "maxWidth": 500,
                  },
                  "variant": "h6",
                },
                "ref": null,
                "rendered": "Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)
                  className={undefined}
                  variant="subtitle1"
>
                  Entre em contato através do 0800 644 1500 se o problema persistir.
</WithStyles(Typography)>,
                "component": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Entre em contato através do 0800 644 1500 se o problema persistir.",
                  "className": undefined,
                  "variant": "subtitle1",
                },
                "ref": null,
                "rendered": "Entre em contato através do 0800 644 1500 se o problema persistir.",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              className={undefined}
              color="primary"
              onClick={[Function]}
>
              Voltar
</WithStyles(Button)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Voltar",
              "className": undefined,
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Voltar",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Dialog)
          BackdropProps={
                    Object {
                              "className": undefined,
                            }
          }
          PaperProps={
                    Object {
                              "className": undefined,
                              "elevation": 2,
                            }
          }
          open={true}
>
          <WithStyles(DialogTitle)
                    disableTypography={true}
          >
                    <WithStyles(Typography)
                              color="error"
                              variant="h4"
                    >
                              Acesso negado
                    </WithStyles(Typography)>
          </WithStyles(DialogTitle)>
          <WithStyles(DialogContent)>
                    <WithStyles(DialogContentText)
                              component="div"
                    >
                              <WithStyles(Typography)
                                        style={
                                                  Object {
                                                            "maxWidth": 500,
                                                          }
                                        }
                                        variant="h6"
                              >
                                        Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.
                              </WithStyles(Typography)>
                    </WithStyles(DialogContentText)>
                    <WithStyles(DialogContentText)
                              component="div"
                    >
                              <WithStyles(Typography)
                                        className={undefined}
                                        variant="subtitle1"
                              >
                                        Entre em contato através do 0800 644 1500 se o problema persistir.
                              </WithStyles(Typography)>
                    </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>
          <WithStyles(DialogActions)>
                    <WithStyles(Button)
                              className={undefined}
                              color="primary"
                              onClick={[Function]}
                    >
                              Voltar
                    </WithStyles(Button)>
          </WithStyles(DialogActions)>
</WithStyles(Dialog)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "BackdropProps": Object {
            "className": undefined,
          },
          "PaperProps": Object {
            "className": undefined,
            "elevation": 2,
          },
          "children": Array [
            <WithStyles(DialogTitle)
              disableTypography={true}
>
              <WithStyles(Typography)
                            color="error"
                            variant="h4"
              >
                            Acesso negado
              </WithStyles(Typography)>
</WithStyles(DialogTitle)>,
            <WithStyles(DialogContent)>
              <WithStyles(DialogContentText)
                            component="div"
              >
                            <WithStyles(Typography)
                                          style={
                                                        Object {
                                                                      "maxWidth": 500,
                                                                    }
                                          }
                                          variant="h6"
                            >
                                          Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.
                            </WithStyles(Typography)>
              </WithStyles(DialogContentText)>
              <WithStyles(DialogContentText)
                            component="div"
              >
                            <WithStyles(Typography)
                                          className={undefined}
                                          variant="subtitle1"
                            >
                                          Entre em contato através do 0800 644 1500 se o problema persistir.
                            </WithStyles(Typography)>
              </WithStyles(DialogContentText)>
</WithStyles(DialogContent)>,
            <WithStyles(DialogActions)>
              <WithStyles(Button)
                            className={undefined}
                            color="primary"
                            onClick={[Function]}
              >
                            Voltar
              </WithStyles(Button)>
</WithStyles(DialogActions)>,
          ],
          "open": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Typography)
                color="error"
                variant="h4"
>
                Acesso negado
</WithStyles(Typography)>,
              "disableTypography": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Acesso negado",
                "color": "error",
                "variant": "h4",
              },
              "ref": null,
              "rendered": "Acesso negado",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(DialogContentText)
                  component="div"
>
                  <WithStyles(Typography)
                                    style={
                                                      Object {
                                                                        "maxWidth": 500,
                                                                      }
                                    }
                                    variant="h6"
                  >
                                    Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.
                  </WithStyles(Typography)>
</WithStyles(DialogContentText)>,
                <WithStyles(DialogContentText)
                  component="div"
>
                  <WithStyles(Typography)
                                    className={undefined}
                                    variant="subtitle1"
                  >
                                    Entre em contato através do 0800 644 1500 se o problema persistir.
                  </WithStyles(Typography)>
</WithStyles(DialogContentText)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Typography)
                    style={
                                        Object {
                                                            "maxWidth": 500,
                                                          }
                    }
                    variant="h6"
>
                    Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.
</WithStyles(Typography)>,
                  "component": "div",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.",
                    "style": Object {
                      "maxWidth": 500,
                    },
                    "variant": "h6",
                  },
                  "ref": null,
                  "rendered": "Você não tem permissão para acessar este recurso, solicite ao responsável, por favor.",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Typography)
                    className={undefined}
                    variant="subtitle1"
>
                    Entre em contato através do 0800 644 1500 se o problema persistir.
</WithStyles(Typography)>,
                  "component": "div",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Entre em contato através do 0800 644 1500 se o problema persistir.",
                    "className": undefined,
                    "variant": "subtitle1",
                  },
                  "ref": null,
                  "rendered": "Entre em contato através do 0800 644 1500 se o problema persistir.",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                className={undefined}
                color="primary"
                onClick={[Function]}
>
                Voltar
</WithStyles(Button)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Voltar",
                "className": undefined,
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Voltar",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
